pkg/expression/exprstatic/exprctx.go

```diff
@@ -299,6 +299,11 @@ func (ctx *ExprContext) GetStaticConvertibleEvalContext() exprctx.StaticConverti
 	return ctx.evalCtx
 }
 
+// IsReadonlyUserVar implements the `BuildContext.IsReadonlyUserVar` and should always return false.
+func (ctx *ExprContext) IsReadonlyUserVar(name string) bool {
+	return false
```

Comment:
Could you add more comments on why we always return false here? I think we can also introduce a option `WithReadyOnlyUserVars([]string)` when constructing `ExprContext`. But for simple, it always return false here and it is safe for most cases.

Reply:
Add some comments. The static context is only used during execution. So we don't need to consider the readonly user var currently